contracts/nexus-gateway/src/contract.rs

```diff
@@ -115,80 +112,70 @@ mod tests {
+            deps.as_mut(),
+            mock_env(),
+            mock_info(
+                AXELARNET_GATEWAY,
+                &[Coin::new(100, "test"), Coin::new(100, "test")],
+            ),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
+        );
+        assert_err_contains!(res, Error, Error::InvalidToken);
+    }
+
+    #[test]
+    fn route_message_with_token() {
         let mut deps = mock_dependencies();
-        deps.querier = deps
-            .querier
-            .with_custom_handler(reply_with_tx_hash_and_nonce(tx_hash, nonce));
-        deps.querier.update_wasm(move |msg| match msg {
-            WasmQuery::Smart { contract_addr, msg } if contract_addr == AXELARNET_GATEWAY => {
-                let msg = from_json::<axelarnet_gateway::msg::QueryMsg>(msg).unwrap();
-
-                match msg {
-                    axelarnet_gateway::msg::QueryMsg::ChainName => {
-                        QuerierResult::Ok(to_json_binary("axelarnet").into())
-                    }
-                    _ => panic!("unexpected query: {:?}", msg),
-                }
-            }
-            _ => panic!("unexpected query: {:?}", msg),
-        });
         instantiate_contract(deps.as_mut());
 
-        let token = Coin {
-            denom: "denom".to_string(),
-            amount: 100u128.into(),
-        };
         let res = execute(
             deps.as_mut(),
             mock_env(),
-            mock_info(NEXUS, &[token.clone()]),
-            ExecuteMsg::CallContractWithToken {
-                destination_chain: "destinationChain".parse().unwrap(),
-                destination_address: "0xD419".parse().unwrap(),
-                payload: decode("bb9b5566c2f4876863333e481f4698350154259ffe6226e283b16ce18a64bcf1")
-                    .unwrap()
-                    .into(),
-            },
+            mock_info(AXELARNET_GATEWAY, &[Coin::new(100, "test")]),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
         );
-        assert!(res.is_ok_and(move |res| match res.messages.as_slice() {
-            [SubMsg {
-                msg: CosmosMsg::Bank(BankMsg::Send { to_address, amount }),
-                ..
-            }, SubMsg {
-                msg:
-                    CosmosMsg::Custom(nexus::execute::Message {
-                        token: Some(actual_token),
-                        ..
-                    }),
-                ..
-            }] => *actual_token == token && to_address == NEXUS && *amount == vec![token],
-            _ => false,
-        }));
+        let res = assert_ok!(res);
+        goldie::assert_json!(res);
+
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info(AXELARNET_GATEWAY, &[Coin::new(100, "test")]),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
+        );
+        let res = assert_ok!(res);
+        assert!(res.messages.is_empty());
```

Comment:
```suggestion
        let response = assert_ok!(execute(
            deps.as_mut(),
            mock_env(),
            mock_info(AXELARNET_GATEWAY, &[Coin::new(100, "test")]),
            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
        ));
        assert!(response.messages.is_empty());
```
Sorry, I meant this, similar to what we do [here](https://github.com/axelarnetwork/axelar-amplifier/pull/624/files#diff-5f32f8b97bf50fa78ba3747e50ecf568d72d30f3e191a3fc20fd98633c76c243R333)